--- work/CMakeLists.txt.old	2018-08-11 06:44:16.514176700 -0500
+++ work/CMakeLists.txt	2018-08-11 06:43:53.760659700 -0500
@@ -1112,7 +1112,7 @@
     testout_3x2_ifast.ppm testout_3x2_ifast_prog.jpg
     ${MD5_PPM_3x2_IFAST} cjpeg-${libtype}-3x2-ifast-prog)
 
-  if(WITH_ARITH_ENC)
+  if(WITH_ARITH_ENC and NOT WITH_12BIT)
     # CC: YCC->RGB  SAMP: fullsize/h2v2  FDCT: islow  ENT: arith
     add_bittest(cjpeg 420-islow-ari "-dct;int;-arithmetic"
       testout_420_islow_ari.jpg ${TESTIMAGES}/testorig.ppm
@@ -1129,7 +1129,7 @@
       ${MD5_JPEG_444_ISLOW_PROGARI})
   endif()
 
-  if(WITH_ARITH_DEC)
+  if(WITH_ARITH_DEC and NOT WITH_12BIT)
     # CC: RGB->YCC  SAMP: h2v2 merged  IDCT: ifast  ENT: arith
     add_bittest(djpeg 420m-ifast-ari "-fast;-ppm"
       testout_420m_ifast_ari.ppm ${TESTIMAGES}/testimgari.jpg
